mmpose/datasets/pipelines/hand_transform.py

```diff
@@ -0,0 +1,198 @@
+        joints_3d_visible = results['joints_3d_visible']
+        cfg = results['ann_info']
+        image_size = cfg['image_size']
+        W, H, D = cfg['heatmap_size']
+        bbox_depth_size = cfg['bbox_depth_size']
+        joint_weights = cfg['joint_weights']
+        use_different_joint_weights = cfg['use_different_joint_weights']
+
+        mu_x = joints_3d[:, 0] * W / image_size[0]
+        mu_y = joints_3d[:, 1] * H / image_size[1]
+        mu_z = (joints_3d[:, 2] / bbox_depth_size + 0.5) * D
+
+        target_weight = joints_3d_visible[:, 0]
+        target_weight = target_weight * (mu_z >= 0) * (mu_z < D)
+        if use_different_joint_weights:
+            target_weight = np.multiply(target_weight, joint_weights)
+        target_weight = target_weight[:, None]
+
+        x, y, z = np.arange(W), np.arange(H), np.arange(D)
+        zz, yy, xx = np.meshgrid(z, y, x)
+        xx = xx[None, ...].astype(np.float32)
+        yy = yy[None, ...].astype(np.float32)
+        zz = zz[None, ...].astype(np.float32)
+
+        mu_x = mu_x[..., None, None, None]
+        mu_y = mu_y[..., None, None, None]
+        mu_z = mu_z[..., None, None, None]
+
+        target = np.exp(-((xx - mu_x)**2 + (yy - mu_y)**2 + (zz - mu_z)**2) /
+                        (2 * self.sigma**2))
+
+        results['target'] = target
+        results['target_weight'] = target_weight
+        return results
+
+
+@PIPELINES.register_module()
+class HandGenerateDepthTarget:
+    """Generate the target relative root depth.
+
+    Required keys: 'rel_root_depth', 'rel_root_valid', 'ann_info'. Modified
+    keys: 'target', 'target_weight'.
+    """
+
+    def __init__(self):
+        pass
+
+    def __call__(self, results):
+        """Generate the target heatmap."""
+        rel_root_depth = results['rel_root_depth']
+        rel_root_valid = results['rel_root_valid']
+        cfg = results['ann_info']
+        D = cfg['heatmap_size_root']
+        depth_size = cfg['bbox_depth_size_root']
+        target = (rel_root_depth / depth_size + 0.5) * D
+        target_weight = rel_root_valid * (target >= 0) * (target <= D)
+        results['target'] = target * np.ones(1, dtype=np.float32)
+        results['target_weight'] = target_weight * np.ones(1, dtype=np.float32)
+        return results
+
+
+@PIPELINES.register_module()
+class HandGenerateLabelTarget:
+    """Generate the target hand type label.
+
+    Required keys: 'hand_type', 'hand_type_valid'. Modified keys: 'target',
+    'target_weight'.
+    """
+
+    def __init__(self):
+        pass
+
+    def __call__(self, results):
+        """Generate the target hand type label."""
+        target = results['hand_type']
+        target_weight = np.ones(
+            target.shape, dtype=np.float32) * results['hand_type_valid']
+        results['target'] = target
+        results['target_weight'] = target_weight
+        return results
+
+
+@PIPELINES.register_module()
+class MultitaskGatherTarget:
```

Comment:
Can we put this into shared_transform.py?

Reply:
fixed